modules/react/common/lib/responsive/useResponsiveContainerStyles.ts

```diff
@@ -0,0 +1,143 @@
+import { useTheme } from "@workday/canvas-kit-react/common";
+import type {
+  FlexStyleProps,
+  GridProps
+} from "@workday/canvas-kit-react/layout";
+
+export const breakpointResKeys = ["zero", "s", "m", "l", "xl"] as const;
+export type BreakpointKeys = typeof breakpointResKeys[number];
+// TODO: export just all the style props
+type AllStyleProps = Omit<GridProps, "children" | "as"> & FlexStyleProps;
+type ResponsiveCSSObject<T> = {
+  [P in keyof T]: Partial<Record<BreakpointKeys, AllStyleProps>> &
+    AllStyleProps;
+};
+type CSSObject<T> = {
+  [P in keyof T]: AllStyleProps;
+};
+
+const isWithinBreakpoint = (width: number, min: number, max?: number) => {
+  if (width >= min && max === undefined) {
+    return true;
+  }
+  if (width >= min && max && width <= max - 1) {
+    return true;
+  }
+  return false;
+};
+
+/**
+ * `useResponsiveContainerStyles` - This is a hook that will allow you to create container-based
+responsive styles with objects (as you can see in the example below). This hook accepts three
+arguments with the third being optional (Style Objects, Container Width, Theme). In each style
+object, there are five sizes that it will accept: `zero, small, medium, large and extra large`.
+These give sizes represent the breakpoints that are at the bottom of this page. The sizes will act
```

Comment:
what do you mean bottom of the page?

Reply:
Adjusted wording. 

I was referring to the table at the bottom of the docs page